myia/parser.py

```diff
@@ -311,13 +381,73 @@ def process_Return(self, block: 'Block', node: ast.Return) -> 'Block':
     def process_Assign(self, block: 'Block', node: ast.Assign) -> 'Block':
         """Process an assignment."""
         anf_node = self.process_node(block, node.value)
-        if len(node.targets) == 1 and isinstance(node.targets[0], ast.Name):
-            anf_node.debug.name = node.targets[0].id
-            block.write(node.targets[0].id, anf_node)
-        else:
-            raise NotImplementedError(node.targets)
+        targ, = node.targets
+
+        def write(targ, anf_node):
+
+            if isinstance(targ, ast.Name):
+                # CASE: x = value
+                anf_node.debug.name = targ.id
+                block.write(targ.id, anf_node)
+
+            elif isinstance(targ, ast.Tuple):
+                # CASE: x, y = value
+                for i, elt in enumerate(targ.elts):
+                    op = self.environment.map(operator.getitem)
+                    new_node = Apply([op, anf_node, Constant(i)], block.graph)
+                    write(elt, new_node)
+
+            elif isinstance(targ, ast.Subscript):
+                if isinstance(targ.value, ast.Name):
+                    # CASE: x[y] = value
+                    op = self.environment.map(operator.setitem)
+                    obj = self.process_node(block, targ.value)
+                    idx = self.process_node(block, targ.slice)
+                    new_node = Apply([op, obj, idx, anf_node], block.graph)
+                    write(targ.value, new_node)
+                else:
+                    # UNSUPPORTED: f()[x] = value
+                    raise NotImplementedError(
+                        "You can only set a slice on a variable."
+                    )  # pragma: no cover
+
+            elif isinstance(targ, ast.Attribute):
+                if isinstance(targ.value, ast.Name):
+                    # CASE: x.y = value
+                    op = self.environment.map(setattr)
+                    obj = self.process_node(block, targ.value)
+                    idx = Constant(targ.attr)
+                    new_node = Apply([op, obj, idx, anf_node], block.graph)
+                    write(targ.value, new_node)
+                else:
+                    # UNSUPPORTED: f().x = value
+                    raise NotImplementedError(
+                        "You can only set an attribute on a variable."
+                    )  # pragma: no cover
```

Comment:
I'm against attribute assignment for the same reasons as subscript assignment above.